src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -89,36 +88,48 @@ CHIP_ERROR OTAImageProcessorImpl::Apply()
 #endif
 }
 
-CHIP_ERROR OTAImageProcessorImpl::ProcessBlock(ByteSpan & block)
+CHIP_ERROR OTAImageProcessorImpl::SwitchToNextImage(const ByteSpan & aRemainingData)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    mCurrentImage.mFileInfo  = &mContentHeader.mFiles[static_cast<uint8_t>(ImageType::kNetImage)];
+    mCurrentImage.mImageType = ImageType::kNetImage;
+
+    if (mCurrentImage.mFileInfo->mFileSize > 0)
+    {
+        // finish app-core dfu target to inform mcuboot that all bytes were written
+        ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_done(true)));
+        // Reset app-core target to allow switching a target to the next one
+        ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_reset()));
+        // initialize next dfu target to store net-core image.
+        ReturnErrorOnFailure(System::MapErrorZephyr(
+            dfu_target_init(DFU_TARGET_IMAGE_TYPE_MCUBOOT, static_cast<uint8_t>(mCurrentImage.mImageType), /* size */ 0, nullptr)));
+        // write remaining data to new image
+        ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_write(aRemainingData.data(), aRemainingData.size())));
+        mCurrentImage.mCurrentOffset = aRemainingData.size();
+    }
+    return error;
```

Comment:
`return CHIP_NO_ERROR` and remove the stack variable?

Reply:
You're right, done.